### DISCUSSION TITLE
Return data and errors

### DISCUSSION THREAD

**Discussion by thomaskoller** [strawberry-graphql/strawberry#2261] [Q&A] [answered]
Hi everybody!

I got a use case I'm not sure how to implement. I'd like to provide the functionality of a multi-delete mutation. 
Now the problem I'm facing is, that for whatever reason some of the items I want to delete can not be deleted.
But I do not want to raise a an exception and abort everything if something goes wrong, but I want to return the items which have been deleted in the data field and the items which couldn't be deleted in the error field of the standard response.
Anybody got an idea how to handle this?

Thanks!
Greetz, Thomas

**Comment by estyxx** [comment, marked as answer]
Hi! 

I think the best approach is to return a Union Type with both Error type and your Object type. 
Here there is an example of what I mean: https://play.strawberry.rocks/?gist=a414da7f71ead354c799b062f8e34965
Since the `errors` field is used when an unexpected exception occurs. 

Let me know if that helps! 
We also have a [Discord](https://strawberry.rocks/discord) channel where we can chat :)

**Reply by thomaskoller** [reply]
Thanks for the quick reply!
A union with an error type is something I haven't thought about but I really like the idea! 😎
Good to know, will definitely join the channel:-)
